src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -480,21 +507,37 @@ private string CreateTokenPrivate(JObject payload, SigningCredentials signingCre
                 header.Merge(JObject.FromObject(additionalHeaderClaims));
 
             var rawHeader = Base64UrlEncoder.Encode(Encoding.UTF8.GetBytes(header.ToString(Formatting.None)));
-
-            if (SetDefaultTimesOnTokenCreation)
+            try
             {
-                var now = EpochTime.GetIntDate(DateTime.UtcNow);
-                if (!payload.TryGetValue(JwtRegisteredClaimNames.Exp, out _))
-                    payload.Add(JwtRegisteredClaimNames.Exp, now + TokenLifetimeInMinutes * 60);
+                var jsonPayload = JObject.Parse(payload);
+                if (SetDefaultTimesOnTokenCreation)
+                {
+                    var now = EpochTime.GetIntDate(DateTime.UtcNow);
+                    if (!jsonPayload.TryGetValue(JwtRegisteredClaimNames.Exp, out _))
+                        jsonPayload.Add(JwtRegisteredClaimNames.Exp, now + TokenLifetimeInMinutes * 60);
 
-                if (!payload.TryGetValue(JwtRegisteredClaimNames.Iat, out _))
-                    payload.Add(JwtRegisteredClaimNames.Iat, now);
+                    if (!jsonPayload.TryGetValue(JwtRegisteredClaimNames.Iat, out _))
+                        jsonPayload.Add(JwtRegisteredClaimNames.Iat, now);
+
+                    if (!jsonPayload.TryGetValue(JwtRegisteredClaimNames.Nbf, out _))
+                        jsonPayload.Add(JwtRegisteredClaimNames.Nbf, now);
+                }
+                payload = jsonPayload.ToString(Formatting.None);
 
-                if (!payload.TryGetValue(JwtRegisteredClaimNames.Nbf, out _))
-                    payload.Add(JwtRegisteredClaimNames.Nbf, now);
+                if (encryptingCredentials != null)
+                {
+                    if (additionalHeaderClaims == null)
+                        additionalHeaderClaims = new Dictionary<string, object> { { JwtHeaderParameterNames.Cty, JwtConstants.HeaderType } };
```

Comment:
if the parameter tokenType is not empty or null, should that be used?

Reply:
If the 'cty' not sent in additionalclaims and tokentype is provided, I think we can use it for 'cty'

Reply:
Discussed offline: tokenType is different from cty.